### ISSUE TITLE
EntityMetadata ignores Schemas for OracleDriver

### ISSUE THREAD

**Issue by jf214** [typeorm/typeorm#4820]
**Issue type:**

[ ] question
[X] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[X] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[X] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I'm using the OracleDriver and after upgrading to the 0.2.19, the generated queries ignore the schemas defined on my Entity Definition.

Example Entity
```ts
import {
  Column,
  Entity,
  PrimaryColumn,
} from "typeorm";

@Entity({
  name: "PERSON",
  schema: "DW_ADMIN"
})
export class Person {
    @PrimaryColumn({ name: "ID" })
    public id: number;

    @Column({ name: "NAME" })
    public name: string;
}
```

The resulting query:
```sql
SELECT "p"."ID" AS "p_ID", "p"."NAME" AS "p_NAME" FROM "PERSON"
```

The result is an oracle error because there is no schema attached to the table in the query
```ORA-00942: table or view does not exist``` 

Prior to upgrading the query was:
```sql
SELECT "p"."ID" AS "p_ID", "p"."NAME" AS "p_NAME" FROM "DW_ADMIN"."PERSON"
```

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
!>

**Comment by marrasen** [comment]
We have a similar problem using MySQL. I think this is a problem for all databases except Postgres and SqlServer since this commit:

https://github.com/typeorm/typeorm/commit/d8f1c81dcfed33bf73493fccd162dff70f2679f9

**Comment by kdi-1** [comment]
The same problem. Any workaround?

**Comment by marrasen** [comment]
> The same problem. Any workaround?

We reverted to version 0.2.18 until this is resolved

**Comment by jf214** [comment]
We've done the same.

**Comment by marrasen** [comment]
@jf214 Just discovered that if you change your entities from saying `schema` to `database` you can avoid this bug

```
@Entity("table_name", {database: "schemaname"})
```

**Comment by jf214** [comment]
Nice workaround. I'll try it out.

**Comment by jadilson12** [comment]
@marrasen 

thanks, it worked

**Comment by imnotjames** [comment]
This was fixed with #6673
